content/development/contribution_guidelines.md

```diff
@@ -365,67 +369,78 @@ Rule tests also provide an excellent way to test WAF engines and implementations
-            Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
-          method: POST
-          port: 80
-          uri: "/"
-          data: "var=` /bin/cat /etc/passwd`"
-          version: HTTP/1.1
-        output:
-          log_contains: id "932230"
+    - input:
+      dest_addr: 127.0.0.1
+      headers:
+        Host: localhost
+        User-Agent: "OWASP CRS test agent"
+        Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
+      method: POST
+      port: 80
+      uri: "/"
+      data: "var=` /bin/cat /etc/passwd`"
+      version: HTTP/1.1
+    output:
+      log:
+        expect_ids: [932230]
 ```
 
-This test will succeed if the log output contains `id "932230"`, which would indicate that the rule in question matched and generated an alert.
+This test will succeed if the log contains an entry for rule 932230, which would indicate that the rule in question matched and generated an alert.
 
 It's important that tests consistently include the HTTP header fields `Host`, `User-Agent`, and `Accept`. CRS includes rules that detect if these headers are missing or empty, so these headers should be included in each test to avoid unnecessarily causing those rules to match. Ideally, *each positive test should cause* **only** *the rule in question to match*.
 
-The rule's description field, `desc`, is important. It should describe what is being tested: what *should* match, what should *not* match, etc.
+The rule's description field, `desc`, is important. It should describe what is being tested: what *should* match, what should *not* match, etc. It is often a good idea to use the YAML literal string scalar form, as it makes it easey to write descriptions spanning multiple lines:
+
+```yaml
+- test_id: 1
+  desc: |
+    This is the first line of the description.
+    On the second line, we might see the payload.
+
+    NOTE: this is something important to take into account.
+  stages:
+  #...
+```
 
 ### Negative Tests
 
 Example of a simple *negative test*:
 
 ```yaml
-- test_title: 932260-4
+- test_id: 4
   stages:
-    - stage:
-        input:
-          dest_addr: "127.0.0.1"
-          method: "POST"
-          port: 80
-          headers:
-            User-Agent: "OWASP CRS test agent"
-            Host: "localhost"
-            Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
-          data: 'foo=ping pong tables'
-          uri: '/'
-        output:
-          no_log_contains: id "932260"
+    - input:
+      dest_addr: "127.0.0.1"
+      method: "POST"
+      port: 80
+      headers:
+        User-Agent: "OWASP CRS test agent"
+        Host: "localhost"
+        Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
+      data: 'foo=ping pong tables'
+      uri: '/'
```

Comment:
`/post`

Reply:
Using HTTP verbs is no longer required. We now use Albedo as a backend, where these verbs have no meaning.